### ISSUE TITLE
buck shared cache capabilities

### ISSUE THREAD

**Issue by AlexDreyfuss** [facebook/buck#2405]
hi,

I have been trying to configure buck to use a shared cache on the network.
when on both machines i put my project on /home/local/bucktest/application
it works great. 
when i change the dir to the project to lets say  /home/local/bucktest1/application the access to the cache fails. probably because the rule key hash changes.

Is there a a way that buck includes only relative paths into the hash computation?
It would be too hard to force developers to use buck on preset dirs and paths?

**Comment by swgillespie** [comment]
> Is there a a way that buck includes only relative paths into the hash computation?

No, this is very much by design; in order for the cache to be truly shared between machines in the presence of tools that occasionally encode their current working directory in their outputs, it would be incorrect to *not* do so. As far as I know there's not an option that would turn this off.

**Comment by AlexDreyfuss** [comment]
so how would you share cache between different users, with different home or root directories?
for example /homes/tom and /homes/bob

Also, out of curiosity, what tools occasionally encode their current working directory into their outputs?

Thanks for the quick response!
